Documentation/contributing/development/dev_setup.rst

```diff
@@ -10,35 +10,52 @@
 Development Setup
 =================
 
-Dev Container
-~~~~~~~~~~~~~
+This page provides an overview of different methods for efficient
+development on Cilium. Depending on your needs, you can choose the most
+suitable method.
 
-Cilium provides `Dev Container <https://code.visualstudio.com/docs/devcontainers/containers>`_ configuration for Visual Studio Code Remote Containers
-and `Github Codespaces <https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers>`_.
-This allows you to use a preconfigured development environment in the cloud or locally.
-The container is based on the official Cilium builder image and provides all the dependencies
-required to build Cilium.
+Quick Start
+-----------
 
-.. note::
+If you're in a hurry, here are the essential steps to get started:
 
-    The current Dev Container is running as root. Non-root user support requires non-root
-    user in Cilium builder image, which is related to :gh-issue:`23217`.
+On Linux:
+
+1. ``make kind`` - Provisions a Kind cluster.
+2. ``make kind-install-cilium-fast`` - Installs Cilium on the Kind cluster.
+3. ``make kind-image-fast`` - Builds Cilium and deploys it.
+
+On any OS:
+
+1. ``make kind`` - Provisions a Kind cluster.
+2. ``make kind-image`` - Builds Docker images.
+3. ``make kind-install-cilium`` - Installs Cilium on the Kind cluster.
+
+Detailed Instructions
+---------------------
+
+Depending on your specific development environment and requirements, you
+can follow the detailed instructions below.
 
 Verifying Your Development Setup
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Assuming you have Go installed, you can quickly verify many elements of your
-development setup by running:
+development setup by running the following command:
 
 .. code-block:: shell-session
 
     $ make dev-doctor
 
-Requirements
-~~~~~~~~~~~~
+Depending on your end-goal, not all dependencies listed are required to develop
+on Cilium. For example, "Ginkgo" is not required if you want to improve our
+documentation. Thus, do not consider that you need to have all tools installed.
```

Comment:
Ideally, the follow-up to this would be to add details on what tasks the deps relate to :innocent:.

Reply:
I agree :sweat_smile: :innocent: